### PR TITLE
Apply PEP-8 compatible code formatting

### PR DESCRIPTION
**Pull request by timhoffm**
While not visible in the cheatsheet itself, we should still adhere to common best practices, to be a role model for potential contributors.

### CODE REVIEW

File: scripts/advanced-plots.py [hunk 1]
```diff
@@ -8,59 +8,57 @@
...
 
 # Step plot
 # -----------------------------------------------------------------------------
 X = np.linspace(0, 10, 16)
-Y = 4+2*np.sin(2*X)
+Y = 4 + 2*np.sin(2*X)
 ax.step(X, Y, color="C1", linewidth=0.75)
-ax.set_xlim(0, 8), ax.set_xticks(np.arange(1,8))
-ax.set_ylim(0, 8), ax.set_yticks(np.arange(1,8))
+ax.set_xlim(0, 8), ax.set_xticks(np.arange(1, 8))
```
Comment (QuLogic) [inline scripts/advanced-plots.py:23]: Could this use `ax.set` now? At the very least, if we're going to stuff multiple statements in one line, we should use `;`, not `,` to fake it.

Comment (timhoffm) [reply scripts/advanced-plots.py:23]: Yes, we have that kind of code in our plot-types gallery. I thought I don't make too many changes at once to keep reviewing simpler.

**Comment by jimustafa** [comment]
Would anyone object to applying [*Black*](https://github.com/psf/black) code formatting? If not, I can open a PR. Whitespace formatting and PEP-8 compliance could be enforced through pre-commit checks.

**Comment by timhoffm** [comment]
We intentionally don't use black in the matplotlib core codebase. And personally, I'm not a big fan of it, but it would be ok for me in this repo.

**Comment by jklymak** [comment]
I'm fine with this as well - `black` seems to fall on its face for complicated code, but I think most of these scripts are simple enough that it will work well.  OTOH, I don't use black day-to-day, so would I need to add a pre-commit hook or something?

**Comment by jklymak** [comment]
a) I'm fine with these, 
b) I'm lazy, but it would be nice to know none of these figures changed due to a typo.  This gets into the same problem the main library has, but I wonder if we should add image comparison tests so we can see if code changes change images, and so we can track those changes over time.

**Comment by jimustafa** [comment]
Actually, I also do not care for it much, but it seems to be the best game in town. Adds a bit too much whitespace around maths expressions, IMHO. Maybe just checking for PEP-8 violations is sufficient.

I started looking at [matplotlib/pytest-mpl](https://github.com/matplotlib/pytest-mpl); @jklymak, this could be a way to do regression testing on the figures and cheatsheets. Ideally, not every tiny change is tracked, and comparisons can be made between the current and baseline versions, maybe with the baseline tracked in a separate branch and replaced on each successful build (like the `gh-pages` branch). This might miss a small drift, but could save on bloating the repo.

**Comment by jklymak** [comment]
Cool - I hope this wouldn't end up too bloated as I don't think there is much call for changing the images often.  OTOH they may change as Matplotlib changes, which may end up being a pain...

**Comment by jklymak** [comment]
BTW, I'd just vote for flake8 as that is less invasive than black, but that is just my (relative) familiarity with flake8 speaking

**Comment by jimustafa** [comment]
I will follow up on this PR with another that adds flake8 and applies most of the fixes for PEP-8 compliance.

**Comment by rougier** [comment]
I would also prefer pep-8 over black because I find black a bit too intrusive.

**Pull request merged**